src/main/kotlin/pl/jutupe/ktor_rabbitmq/RabbitMQ.kt

```diff
@@ -35,6 +43,12 @@ class RabbitMQ(
     fun <T> serialize(body: T): ByteArray =
         configuration.serializeBlock.invoke(body!!)
 
+    override fun close() {
+        logger?.info("RabbitMQ stopping")
+        shutdownExecutor.awaitTermination(configuration.shutdownTimeout, TimeUnit.MILLISECONDS)
```

Comment:
you are not invoking shutdown first, though. See https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html
from my understanding one should first shutdown, and then await termination.

Reply:
we should revise this pr

Reply:
Yeah, sure. I'll give it another try on ktor 2.0